e2e/async.go

```diff
@@ -0,0 +1,90 @@
+package e2e
+
+import (
+	"fmt"  //nolint:goimports
+	"sync" //nolint:goimports
+
+	snapapi "github.com/kubernetes-csi/external-snapshotter/client/v4/apis/volumesnapshot/v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2elog "k8s.io/kubernetes/test/e2e/framework/log"
+)
+
+type configItem struct {
+	ID         int // Unique identification
+	f          *framework.Framework
+	obj        *objects // Objects which are set for callback
+	callbackFn string
+	retErr     chan error // error channel to receive errors
+}
+
+type objects struct {
+	snap *snapapi.VolumeSnapshot
+	pvc  *v1.PersistentVolumeClaim
+	app  *v1.Pod
+}
+
+func executeOperations(ch <-chan configItem) {
+	for proc := range ch {
+		switch proc.callbackFn {
```

Comment:
`callbackFn` should probably just be a `func()` and get executed. Coding of the async operations should all be at the same place where the test is executed, not spread out over different files.